### ISSUE TITLE
security vulnerability using  socks@2.7.1

### ISSUE THREAD

**Issue by yonayarin** [raineorshine/npm-check-updates#1408]
- [] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [] I am using the latest version of `npm-check-updates`
- [] I am using `node >= 14.14`

---
Lately I started to get some security vulnerability from this package.

└─┬ npm-check-updates@16.14.20
  └─┬ make-fetch-happen@11.1.1
    └─┬ socks-proxy-agent@7.0.0
       └─┬ socks@2.7.1
         └── ip@2.0.0
      
socks package in this version uses "ip" version "2.0.0" - full issue description https://github.com/nodejs/node/issues/51848
Here is another report of this issue - https://github.com/npm/cli/issues/7223

Will be happy if you can update versions accordingly to remove this issued dependency.
## Steps to Reproduce

Steps:

Run CI with npm-check-updates@16.14.20 installed

## Current Behavior

Display security vulnerability.

**Comment by raineorshine** [comment]
Thanks for reporting. I added it to `overrides` since the patch has not yet trickled up the dependency chain.

It will be published in the next release, which is currently blocked by #1404.

**Comment by wilhen01** [comment]
Bumping this - version 12.0.0 and later of `make-fetch-happen` removes the `socks-proxy-agent` so should resolve this by getting rid of `ip` in the chain entirely.

**Comment by raineorshine** [comment]
Thanks. I don't think #1404 is going to happen, so I just need to do a major release with what we have now.

I'm traveling next week, but will make some time soon.

**Comment by raineorshine** [comment]
Published in `v17.0.0`.

Note that the override is still needed, because `ip` is still in the dependency tree:

```
npm-check-updates@17.0.0 /Users/raine/projects/npm-check-updates
└─┬ npm-registry-fetch@17.1.0
  └─┬ make-fetch-happen@13.0.0
    └─┬ @npmcli/agent@2.2.0
      └─┬ socks-proxy-agent@8.0.2
        └─┬ socks@2.7.1
          └── ip@2.0.1 overridden
```
